test/urlcat.ts

```diff
@@ -85,22 +85,71 @@ describe('urlcat', () => {
     expect(actual).to.equal(expected);
   });
 
-  it ('Provides an overload (baseUrl, pathTemplate) that works correctly', () => {
+  it('Provides an overload (baseUrl, pathTemplate) that works correctly', () => {
     const expected = 'http://example.com/path';
     const actual = urlcat('http://example.com/', '/path');
     expect(actual).to.equal(expected);
   });
 
-  it ('Handles "//" path correctly (reproduces #7)', () => {
+  it('Handles "//" path correctly (reproduces #7)', () => {
     const expected = 'http://example.com//';
     const actual = urlcat('http://example.com/', '//');
     expect(actual).to.equal(expected);
   });
 
-  it ('Provides an overload (baseTemplate, params) that works correctly', () => {
+  it('Provides an overload (baseTemplate, params) that works correctly', () => {
     const expected = 'http://example.com/path/a%20b?q=b+c';
     const actual = urlcat('http://example.com/path/:p', { p: 'a b', q: 'b c' });
     expect(actual).to.equal(expected);
   });
 
+  it('Can handle allowed Type boolean (true) as param', () => {
+    const expected = 'http://example.com/path/true';
+    const actual = urlcat('http://example.com/path/:p', { p: true });
+    expect(actual).to.equal(expected);
+  });
+
+  it('Can handle allowed Type boolean (false) as param', () => {
+    const expected = 'http://example.com/path/false';
+    const actual = urlcat('http://example.com/path/:p', { p: false });
+    expect(actual).to.equal(expected);
+  });
+
+
+  it('Can handle allowed Type Number as param', () => {
+    const expected = 'http://example.com/path/456';
+    const actual = urlcat('http://example.com/path/:p', { p: 456 });
+    expect(actual).to.equal(expected);
+  });
+
+  it('Can handle allowed Type String as param', () => {
+    const expected = 'http://example.com/path/test';
+    const actual = urlcat('http://example.com/path/:p', { p: 'test' });
+    expect(actual).to.equal(expected);
+  });
+
+  it('Can\'t handle object as param', () => {
+    expect(() => urlcat('http://example.com/path/:p', { p: {} })).to.throw(TypeError, "Path parameter p cannot be of type object.Allowed types are: boolean, string, number.");
```

Comment:
Lines like this are a bit too long to be readable. I would suggest inserting a line break before `.to.throw(...`.